summarizer/src/summarizers/naive.rs

```diff
@@ -0,0 +1,278 @@
+
+        for (i, sentence) in all_sentences.iter().enumerate() {
+            if sentence.len() > min_length && sentence.len() < max_length {
+                let mut words: HashSet<String> = HashSet::new();
+                if ngrams {
+                    for n in 7..15 {
+                        let ngrams = sentence
+                            .chars()
+                            .collect::<Vec<char>>()
+                            .windows(n)
+                            .map(|x| x.iter().collect::<String>())
+                            .collect::<Vec<String>>();
+                        words.extend(ngrams);
+                    }
+                } else {
+                    words = sentence
+                        .split_whitespace()
+                        .map(std::string::ToString::to_string)
+                        .collect::<HashSet<_>>();
+                }
+                let outgoing_connections = HashMap::new();
+                let sentence = Sentence {
+                    words,
+                    index: i,
+                    length: sentence.len(),
+                    outgoing_connections: Some(outgoing_connections),
+                    text: (*sentence).to_string(),
+                    number_of_connections: 0.0,
+                };
+                sentences.lock().unwrap().insert(i, sentence.clone());
+            }
+        }
+        let final_sentences = sentences.lock().unwrap().clone();
+
+        Summariser {
+            bias_strength,
+            sentences: final_sentences,
+            matrix: None,
+            bias_list: HashSet::new(),
+            _marker: std::marker::PhantomData,
+        }
+    }
+
+    fn from_sentences_direct(sentences: HashMap<usize, Sentence>) -> Summariser<'a> {
+        Summariser {
+            sentences,
+            matrix: None,
+            bias_list: HashSet::new(),
+            bias_strength: None,
+            _marker: std::marker::PhantomData,
+        }
+    }
+
+    #[allow(clippy::too_many_arguments, clippy::too_many_lines)]
+    fn top_sentences(
+        &mut self,
+        number_of_sentences_to_return: usize,
+        return_summaries_for_each: bool,
+        chunk_size: Option<usize>,
+        force_sum_all: bool,
+        length_penalty: f32,
+        force_chunk: bool,
+        density: f32,
+        bias_list: Option<HashSet<String>>,
+        bias_strength: Option<f32>,
+    ) -> Vec<Sentence> {
+        //If longer than 10,000, then divide it into portions of 5000 each. Instantiate new Summarisers and call Summariser::from_sentences_direct on each one, passing in the portion of the original sentences (convert the HashMap to a vec). Then call Summariser::top_sentences on each one, passing in the number of sentences to return. Collect the sentences, and pass them to a new instance of Summariser::from_sentences_direct. Then call Summariser::top_sentences on that instance, passing in the number of sentences to return. Return the result.
```

Comment:
:art: linebreaks for readability in IDEs
```suggestion
        // If longer than 10,000, then divide it into portions of 5000 each. Instantiate new
        // Summarisers and call Summariser::from_sentences_direct on each one, passing in the
        // portion of the original sentences (convert the HashMap to a vec). Then call
        // Summariser::top_sentences on each one, passing in the number of sentences to return.
        // Collect the sentences, and pass them to a new instance of
        // Summariser::from_sentences_direct. Then call Summariser::top_sentences on that
        // instance, passing in the number of sentences to return. Return the result.
```